lib/jekyll/document.rb

```diff
@@ -262,12 +271,16 @@ def published?
     def read(opts = {})
       Jekyll.logger.debug "Reading:", relative_path
 
-      if yaml_file?
+      if yaml_file? && !render_yaml_file?
```

Comment:
could change `!render_yaml_file?` to `!write?` so it makes more sense that we get to the else statement.

In the case that we make it to the `else` statement, we merely only need to check that we have a yaml file to know if we should process it properly since this line already determines that we cannot get to the `else` without expecting to read and merge the data with defaults.